propulsion-hotel/Reactor/Handler.fs

```diff
@@ -0,0 +1,88 @@
+module Reactor.Handler
+
+open Infrastructure
+open Propulsion.Internal
+
+[<RequireQualifiedAccess>]
+type Outcome = Completed | Deferred | Resolved of successfully : bool
+
+/// Gathers stats based on the outcome of each Span processed, periodically including them in the Sink summaries
+type Stats(log, statsInterval, stateInterval, ?logExternalStats) =
+    inherit Propulsion.Streams.Stats<Outcome>(log, statsInterval, stateInterval)
+
+    let mutable completed, deferred, failed, succeeded = 0, 0, 0, 0
+
+    override _.DumpStats() =
+        base.DumpStats()
+        if completed <> 0 || deferred <> 0 || failed <> 0 || succeeded <> 0 then
+            log.Information(" Completed {completed} Deferred {deferred} Failed {failed} Succeeded {succeeded}", completed, deferred, failed, succeeded)
+            completed <- 0; deferred <- 0; failed <- 0; succeeded <- 0
+        match logExternalStats with None -> () | Some f -> f Serilog.Log.Logger
+
+    override _.HandleOk res = res |> function
+        | Outcome.Completed -> completed <- completed + 1
+        | Outcome.Deferred -> deferred <- deferred + 1
+        | Outcome.Resolved successfully -> if successfully then succeeded <- succeeded + 1 else failed <- failed + 1
+    override _.Classify(exn) =
+        match exn with
+        | Equinox.DynamoStore.Exceptions.ProvisionedThroughputExceeded -> Propulsion.Streams.OutcomeKind.RateLimited
+        | x -> base.Classify x
+    override _.HandleExn(log, exn) =
+        log.Information(exn, "Unhandled")
+
+open Domain
+
+let private isReactionStream = function
+    | GroupCheckoutProcess.Category -> true
+    | _ -> false
+
+let private handleAction (guestStays : GuestStayAccount.Service) groupCheckoutId (act : GroupCheckoutProcess.Flow.Action) = async {
+    match act with
+    | GroupCheckoutProcess.Flow.Checkout pendingStays ->
+        let attempt stayId groupCheckoutId = async {
+            match! guestStays.GroupCheckout(stayId, groupCheckoutId) with
+            | GuestStayAccount.Decide.GroupCheckoutResult.Ok r -> return Choice1Of2 (stayId, r) 
+            | GuestStayAccount.Decide.GroupCheckoutResult.AlreadyCheckedOut -> return Choice2Of2 stayId } 
+        let! outcomes = seq { for x in pendingStays do attempt x groupCheckoutId } |> Async.parallelThrottled 5
```

Comment:
You could turn the argument order around and make this a simple map. Not sure if that's any better though 😄 

```suggestion
        let! outcomes = pendIngStays |> Seq.map (attempt groupCheckoutId) |> Async.parallelThrottled 5
```